src/Microsoft.Identity.Client/Internal/ClientCredentialWrapper.cs

```diff
@@ -33,6 +33,8 @@ public ClientCredentialWrapper(ClientAssertionCertificateWrapper certificate)
         internal long ValidTo { get; set; }
         internal bool ContainsX5C { get; set; }
         internal string Audience { get; set; }
+        internal string Secret { get; private set; }
+        internal ClientAssertion UserProvidedClientAssertion { get; set; }
```

Comment:
Can there be a non-user provided ClientAssertion? If not, just use ClientAssertion as the name of the property